### DISCUSSION TITLE
Google Colab best.pt --->Jupyter notebook issue

### DISCUSSION THREAD

**Discussion by rajnitish** [ultralytics/ultralytics#6538] [General]
Dear glenn-jocher
I have trained my model on Yolov8 on google colab. And stored results in project directory. now hen i am trying to trained with new images using last.pt on jupyter notebook. it is giving me error. from ultralytics import YOLO
import os

CURRENT_PATH = os.getcwd()
print(CURRENT_PATH)
custom_checkpoint_path = os.path.join(CURRENT_PATH,"runs","detect","Military_Aircraft_Detection","weights", "last.pt")
model = YOLO(custom_checkpoint_path)

yaml_path = os.path.join(CURRENT_PATH, "Aircraft.yaml")
os.path.join(CURRENT_PATH,"runs","detect","Military_Aircraft_Detection"),
results = model.train(data=yaml_path,name = "ABC",resume=True)

D:\Capstone\new\Capstone_MAD_Mavericks
New https://pypi.org/project/ultralytics/8.0.216 available Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.0.188 Python-3.8.13 torch-2.0.1+cpu CPU (Intel Core(TM) i7-8550U 1.80GHz)
engine\trainer: task=detect, mode=train, model=D:\Capstone\new\Capstone_MAD_Mavericks\runs\detect\Military_Aircraft_Detection\weights\last.pt, data=D:\Capstone\new\Capstone_MAD_Mavericks\Aircraft.yaml, epochs=75, patience=3, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=/content/drive/MyDrive/Capstone_MAD_Mavericks/runs/detect, name=Military_Aircraft_Detection, exist_ok=True, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, stream_buffer=False, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.0, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=\content\drive\MyDrive\Capstone_MAD_Mavericks\runs\detect\Military_Aircraft_Detection
TensorBoard: Start with 'tensorboard --logdir \content\drive\MyDrive\Capstone_MAD_Mavericks\runs\detect\Military_Aircraft_Detection', view at http://localhost:6006/
UsageError Traceback (most recent call last)
Input In [30], in <cell line: 11>()
9 yaml_path = os.path.join(CURRENT_PATH, "Aircraft.yaml")
10 #, exist_ok= True, project = os.path.join(CURRENT_PATH,"Runs","detect"),save_dir= os.path.join(CURRENT_PATH,"runs","detect","Military_Aircraft_Detection"),
---> 11 results = model.train(data=yaml_path,name = "ABC",resume=True)

File C:\anaconda3\lib\site-packages\ultralytics\engine\model.py:334, in Model.train(self, trainer, **kwargs)
332 self.model = self.trainer.model
333 self.trainer.hub_session = self.session # attach optional HUB session
--> 334 self.trainer.train()
335 # Update model and cfg after training
336 if RANK in (-1, 0):

File C:\anaconda3\lib\site-packages\ultralytics\engine\trainer.py:195, in BaseTrainer.train(self)
192 ddp_cleanup(self, str(file))
194 else:
--> 195 self._do_train(world_size)

File C:\anaconda3\lib\site-packages\ultralytics\engine\trainer.py:293, in BaseTrainer._do_train(self, world_size)
291 if world_size > 1:
292 self._setup_ddp(world_size)
--> 293 self._setup_train(world_size)
295 self.epoch_time = None
296 self.epoch_time_start = time.time()

File C:\anaconda3\lib\site-packages\ultralytics\engine\trainer.py:215, in BaseTrainer._setup_train(self, world_size)
210 """
211 Builds dataloaders and optimizer on correct rank process.
212 """
214 # Model
--> 215 self.run_callbacks('on_pretrain_routine_start')
216 ckpt = self.setup_model()
217 self.model = self.model.to(self.device)

File C:\anaconda3\lib\site-packages\ultralytics\engine\trainer.py:160, in BaseTrainer.run_callbacks(self, event)
158 """Run all existing callbacks associated with a particular event."""
159 for callback in self.callbacks.get(event, []):
--> 160 callback(self)

File C:\anaconda3\lib\site-packages\ultralytics\utils\callbacks\wb.py:29, in on_pretrain_routine_start(trainer)
27 def on_pretrain_routine_start(trainer):
28 """Initiate and start project if module is present."""
---> 29 wb.run or wb.init(project=trainer.args.project or 'YOLOv8', name=trainer.args.name, config=vars(trainer.args))

File C:\anaconda3\lib\site-packages\wandb\sdk\wandb_init.py:1184, in init(job_type, dir, config, project, entity, reinit, tags, group, name, notes, magic, config_exclude_keys, config_include_keys, anonymous, mode, allow_val_change, resume, force, tensorboard, sync_tensorboard, monitor_gym, save_code, id, settings)
1182 if logger is not None:
1183 logger.exception(str(e))
-> 1184 raise e
1185 except KeyboardInterrupt as e:
1186 assert logger

File C:\anaconda3\lib\site-packages\wandb\sdk\wandb_init.py:1161, in init(job_type, dir, config, project, entity, reinit, tags, group, name, notes, magic, config_exclude_keys, config_include_keys, anonymous, mode, allow_val_change, resume, force, tensorboard, sync_tensorboard, monitor_gym, save_code, id, settings)
1159 try:
1160 wi = _WandbInit()
-> 1161 wi.setup(kwargs)
1162 assert wi.settings
1163 except_exit = wi.settings._except_exit

File C:\anaconda3\lib\site-packages\wandb\sdk\wandb_init.py:320, in _WandbInit.setup(self, kwargs)
317 # get status of code saving before applying user settings
318 save_code_pre_user_settings = settings.save_code
--> 320 settings._apply_init(kwargs)
321 if not settings._offline and not settings._noop:
322 user_settings = self._wl._load_user_settings()

File C:\anaconda3\lib\site-packages\wandb\sdk\wandb_settings.py:1818, in Settings._apply_init(self, init_settings)
1815 init_settings["resume"] = "auto"
1817 # update settings
-> 1818 self.update(init_settings, source=Source.INIT)
1820 # handle auto resume logic
1821 if self.resume == "auto":

File C:\anaconda3\lib\site-packages\wandb\sdk\wandb_settings.py:1458, in Settings.update(self, settings, source, **kwargs)
1456 for key in self.__modification_order:
1457 if key in settings:
-> 1458 self.dict[key].update(settings.pop(key), source=source)
1460 # update the remaining properties
1461 for key, value in settings.items():

File C:\anaconda3\lib\site-packages\wandb\sdk\wandb_settings.py:579, in Property.update(self, value, source)
561 # - always update value if source == Source.OVERRIDE
562 # - if not previously overridden:
563 # - update value if source is lower than or equal to current source and property is policy
564 # - update value if source is higher than or equal to current source and property is not policy
565 if (
566 (source == Source.OVERRIDE)
567 or (
(...)
577 ):
578 # self.dict["_value"] = self._validate(self._preprocess(value))
--> 579 self._value = self._validate(self._preprocess(value))
580 self._source = source

File C:\anaconda3\lib\site-packages\wandb\sdk\wandb_settings.py:549, in Property._validate(self, value)
545 _validator = (
546 [self._validator] if callable(self._validator) else self._validator
547 )
548 for v in _validator:
--> 549 if not v(value):
550 # failed validation will likely cause a downstream error
551 # when trying to convert to protobuf, so we raise a hard error
552 raise SettingsValidationError(
553 f"Invalid value for property {self.name}: {value}."
554 )
555 return value

File C:\anaconda3\lib\site-packages\wandb\sdk\wandb_settings.py:961, in Settings._validate_project(value)
959 invalid_chars = {char for char in invalid_chars_list if char in value}
960 if invalid_chars:
--> 961 raise UsageError(
962 f"Invalid project name {value!r}: "
963 f"cannot contain characters {','.join(invalid_chars_list)!r}, "
964 f"found {','.join(invalid_chars)!r}"
965 )
966 return True

UsageError: Invalid project name '/content/drive/MyDrive/Capstone_MAD_Mavericks/runs/detect': cannot contain characters '/,\,#,?,%,:', found '/'

**Comment by pderrenger** [comment]
@rajnitish it seems like you're encountering an issue with the project name containing invalid characters when using the `wandb` library. The error message indicates that the project name cannot contain certain characters such as `/`, which is present in your project path.

To resolve this issue, you should specify a valid project name that does not contain any of the restricted characters. Here's how you can modify your code:

```python
from ultralytics import YOLO
import os

CURRENT_PATH = os.getcwd()
print(CURRENT_PATH)
custom_checkpoint_path = os.path.join(CURRENT_PATH, "runs", "detect", "Military_Aircraft_Detection", "weights", "last.pt")
model = YOLO(custom_checkpoint_path)

yaml_path = os.path.join(CURRENT_PATH, "Aircraft.yaml")
project_name = "Military_Aircraft_Detection"  # Use a valid project name without invalid characters
results = model.train(data=yaml_path, name="ABC", project=project_name, resume=True)
```

Make sure that the `project` parameter in the `model.train()` method is set to a string that does not include any of the restricted characters. This should allow you to resume training without encountering the `UsageError` from `wandb`. If you're not using `wandb` for logging, you can also disable it by setting the `--wandb` flag to `None` when initializing your training.
